feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { Product, ProductScope, ProductService } from '@spartacus/core';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { Configurator } from '../../../core/model/configurator.model';
+import { QuantityUpdateEvent } from '../../form/configurator-form.event';
+import { ConfiguratorPriceComponentOptions } from '../../price/configurator-price.component';
+import {
+  ConfiguratorAttributeQuantityComponentOptions,
+  Quantity,
+} from '../quantity/configurator-attribute-quantity.component';
+
+export interface ProductExtended extends Product {
+  noLink?: boolean;
+}
+
+export interface ConfiguratorAttributeProductCardComponentOptions {
+  /** If set to `true`, all action buttons will be disabled.  */
+  disableAllButtons?: boolean;
+  /** If set to `true`, the remove/deselect button won't be available. Usefull for required attributes,
+   *  where a deselect/remove of last value shall not be possible.  */
+  hideRemoveButton?: boolean;
+  multiSelect?: boolean;
+  productBoundValue?: Configurator.Value;
+  singleDropdown?: boolean;
+  withQuantity?: boolean;
+  loading$?: Observable<boolean>;
+  attributeId: number;
+}
+
+@Component({
+  selector: 'cx-configurator-attribute-product-card',
+  templateUrl: './configurator-attribute-product-card.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorAttributeProductCardComponent implements OnInit {
+  product$: Observable<ProductExtended>;
+  loading$ = new BehaviorSubject<boolean>(true);
+
+  @Input()
+  productCardOptions: ConfiguratorAttributeProductCardComponentOptions;
+
+  @Output() handleDeselect = new EventEmitter<string>();
+  @Output() handleQuantity = new EventEmitter<QuantityUpdateEvent>();
+  @Output() handleSelect = new EventEmitter<string>();
+
+  constructor(protected productService: ProductService) {}
+
+  ngOnInit() {
+    this.loading$.next(true);
+    const productSystemId = this.productCardOptions?.productBoundValue
+      ?.productSystemId;
+
+    this.product$ = this.productService
+      .get(productSystemId ? productSystemId : '', ProductScope.LIST)
+      .pipe(
+        map((respProduct) => {
+          return respProduct
+            ? respProduct
+            : this.transformToProductType(
+                this.productCardOptions?.productBoundValue
+              );
+        }),
+        tap(() => this.loading$.next(false))
```

Comment:
this `loading$` is very confusing. It indicates `configuration` loading?

Reply:
pick